app/code/Magento/Theme/Model/View/Design.php

```diff
@@ -138,7 +138,7 @@ public function setDesignTheme($theme, $area = null)
     {
         if ($area) {
             $this->setArea($area);
-        } else {
+        } elseif (null == $area) {
```

Comment:
could you explain why it's needed? Also not clear how it worked before

Reply:
I think we should avoid use `else` in this case -> so go with elseif for check in case equal null

Reply:
@mrtuvn What about situation when custom module pass empty string as `$area` for example?